pkg/daemons/agent/agent.go

```diff
@@ -64,16 +63,16 @@ func startKubeProxy(ctx context.Context, cfg *daemonconfig.Agent) error {
 func startKubelet(ctx context.Context, cfg *daemonconfig.Agent) error {
 	argsMap, defaultConfig, err := kubeletArgsAndConfig(cfg)
 	if err != nil {
-		return pkgerrors.WithMessage(err, "prepare default configuration drop-in")
+		return pkgerrors.WithMessage(err, "prepare default kubelet configuration")
 	}
 
-	extraArgs, err := extractConfigArgs(cfg.KubeletConfigDir, cfg.ExtraKubeletArgs, defaultConfig)
-	if err != nil {
-		return pkgerrors.WithMessage(err, "prepare user configuration drop-ins")
+	if err := writeKubeletConfig(cfg.KubeletConfig, defaultConfig); err != nil {
+		return pkgerrors.WithMessage(err, "generate default kubelet configuration")
```

Comment:
Should we remove the old drop-in at `"00-"+version.Program+"-defaults.conf"` if it exists? If we're going to generate our defaults in a new path, we should clean up the file generated by previous releases to avoid leaving unwanted config overrides behind that will override what is currently expected.

Reply:
no, things do not get cleaned out of there when upgrading. We intentionally leave the kubelet config dir alone so that users can put their own stuff in it.